src/ess/spectroscopy/indirect/ki.py

```diff
@@ -236,155 +125,27 @@ def primary_path_length(
-
-    Parameters
-    ----------
-    file:
-        The HDF5 NeXus file with information about the primary spectrometer
-        under 'entry/instrument'
-    source:
-        The name of the source component in the NeXus instrument group
-    sample:
-        The name of the sample component in the NeXus instrument group
-    frequency:
-        The frequency of the source, e.g., scipp.scalar(14.0, unit='Hz') for ESS
-    duration:
-        The source pulse duration, e.g., approximately scipp.scalar(3.0, unit='msec')
-        for ESS
-    delay:
-        The velocity dependent time delay between source pulse time and neutrons
-        exiting the moderator, should be 1-D (with dim='wavelength') and have values
-        corresponding to the provided `velocities`
-    velocities:
-        The velocities of neutrons produced by the source; must be 1-D
-        (with dim='wavelength') with at least two values. The velocities and delay
-        times should be the same length, with a 1:1 correspondence. These values are
-        then used to define the time-slowness phase space polygon produced by
-        each source pulse.
-
-    Returns
-    -------
-    :
-        A choppera.PrimarySpectrometer object representing the NeXus file contents
-    """
-    from choppera.nexus import primary_spectrometer
-    from scippnexus import File
-
-    with File(file) as data:
-        instrument = data['entry/instrument']
-        if source not in instrument:
-            raise KeyError(f"The source '{source}' is not in the instrument group")
-        if sample not in instrument:
-            raise KeyError(f"The sample '{sample}' is not in the instrument group")
-        return primary_spectrometer(
-            instrument, source, sample, frequency, duration, delay, velocities
-        )
-
-
-def primary_pivot_time(primary: PrimarySpectrometerObject) -> SourceSampleFlightTime:
-    """Return a time, relative to the source pulse time, between neutron-arrival-periods
-
-    Notes
-    -----
-    The determined pivot time is be before the earliest arrival time of neutrons
-    produced in a source pulse that then are transmitted through the primary
-    spectrometer, while also being after the latest arrival time from the preceding
-    pulse.
-
-    Parameters
-    ----------
-    primary:
-        The choppera.PrimarySpectrometer object
-
-    Returns
-    -------
-    :
-        The determined pivot time
-    """
-    from choppera.nexus import primary_pivot_time as primary_time
-
-    return primary_time(primary)
-
-
 def unwrap_sample_time(
-    times: SampleFrameTime, frequency: SourceFrequency, least: SourceSampleFlightTime
-) -> SampleTime:
+    sample_events: SampleEvents,
+    table: TimeOfFlightLookupTable,
+    l1: SourceSamplePathLength,
+) -> TofSampleEvents:
     """Use the pivot time to shift neutron event time offsets, recovering 'real'
     time after source pulse per event"""
```

Comment:
The docstring talks about pivot time, which I'm assuming was from Greg's code. We should update with a better description?